### PR TITLE
feat: Add intents to Switch toggles

### LINKED ISSUE #506: [Components] Switch integration - Specificities
Add missing  Intents

### PR DESCRIPTION
**Pull request by MarinaRomanova**
## 📋 Changes description

Add missing ToggleIntents support to switch toggle component.

## 📸 Screenshots

Present in the current PR

## 🗒️ Other info

close #506

### CODE REVIEW

File: spark/src/main/kotlin/com/adevinta/spark/components/toggles/CheckBox.kt [hunk 1]
```diff
@@ -35,6 +35,7 @@ import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.tooling.preview.PreviewParameter
 import com.adevinta.spark.InternalSparkApi
 import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.components.IntentColor
```
Comment (ThibaultChariere) [inline spark/src/main/kotlin/com/adevinta/spark/components/toggles/CheckBox.kt:38]: Is this import unused ?

**Review by ThibaultChariere — APPROVED** [review]

File: spark/src/main/kotlin/com/adevinta/spark/components/toggles/CheckBox.kt (hunk 1 above)
Comment (MarinaRomanova) [reply spark/src/main/kotlin/com/adevinta/spark/components/toggles/CheckBox.kt]: I ran spotlessApply, I think it was removed :-)

File: spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md [hunk 2]
```diff
@@ -41,6 +41,18 @@ SwitchLabelled(
 ) { Text(text = "Switch On") }
 ```
 
+### Styles
+
+The `Switch` and `SwitchLabelled` accept 7 [ToggleIntent](ToggleIntent.kt)s:
+
+- Primary (default)
```
Comment (ThibaultChariere) [inline spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md:48]: Isn't the `Basic` intent the default one now ?

Comment (MarinaRomanova) [reply spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md:48]: Ah yes, good catch. I lost this change in some complicated rebase

File: spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md [hunk 3]
```diff
@@ -41,6 +41,18 @@ SwitchLabelled(
...
+- Secondary
+- Success
+- Alert
+- Danger
+- Info
+- Neutral
```
Comment (soulcramer) [inline spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md]: We should change the doc as well
```suggestion
- Main 
- Support
- Accent
- Basic (default)
- Success
- Alert
- Danger
- Info
- Neutral
```

**Review by ThibaultChariere — APPROVED** [review]

File: spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md (hunk 3 above)
Comment (MarinaRomanova) [reply spark/src/main/kotlin/com/adevinta/spark/components/toggles/Switch.md]: Yes, I have corrected this in the second commit

**Pull request merged**